person.rb

```diff
@@ -0,0 +1,80 @@
+# person.rb
+
+#
+# Interface Class Nameable, for decorator design
+#
+class Nameable
+  def correct_name
+    raise NotImplementedError, "#{self.class} has not implemented method '#{__method__}'"
+  end
+end
```

Comment:
- [ ] [OPTIONAL] It would be nice if you put this in a separate file to make your code base readable